lnwallet/channel.go

```diff
@@ -3092,6 +3283,89 @@ func (lc *LightningChannel) createCommitDiff(
 	}, nil
 }
 
+// getDanglingRemoteUpdates returns all remote log updates that we haven't
+// signed for yet ourselves.
+func (lc *LightningChannel) getDanglingRemoteUpdates() []channeldb.LogUpdate {
+	// First, we need to convert the funding outpoint into the ID that's
+	// used on the wire to identify this channel.
+	chanID := lnwire.NewChanIDFromOutPoint(&lc.channelState.FundingOutpoint)
+
+	// Fetch the last remote update that we have added to our remote
+	// commitment chain.
+	chainTail := lc.remoteCommitChain.tip()
+	lastCommited := chainTail.theirMessageIndex
+
+	// We'll now run through the remote update log to locate the items that
+	// we haven't signed for yet. This will be the set of items we need to
+	// restore if we reconnect in order to produce the signature that the
+	// remote party expects.
+	var logUpdates []channeldb.LogUpdate
+	for e := lc.remoteUpdateLog.Front(); e != nil; e = e.Next() {
+		pd := e.Value.(*PaymentDescriptor)
+
+		// Skip all remote updates that we have already included in our
+		// commit chain. At the moment this function is called, we've
+		// received a signature from the remote party for all prior
+		// updates.
+		if pd.LogIndex < lastCommited {
```

Comment:
We should only include updates which have a log index that're less than or equal to what's committed on our commitment, and greater than their current message index.

Reply:
See above